scripts/sct_extract_metric.py

```diff
@@ -352,70 +398,29 @@ def main(fname_data, path_label, method, slices, levels, fname_output, labels_us
-    if '-append' in arguments:
-        append_csv = int(arguments['-append'])
-    else:
-        append_csv = 0
-    if '-combine' in arguments:
-        combine_labels = arguments['-combine']
-    else:
-        combine_labels = 0
-    if '-l' in arguments:
-        labels_user = arguments['-l']
-    else:
-        labels_user = ''
-    if '-param' in arguments:
-        adv_param_user = arguments['-param']
-    else:
-        adv_param_user = ''
-    if '-z' in arguments:
-        slices_of_interest = arguments['-z']
-    else:
-        slices_of_interest = ''
-    if '-vert' in arguments:
-        vertebral_levels = arguments['-vert']
-    else:
-        vertebral_levels = ''
-    if '-vertfile' in arguments:
-        fname_vertebral_labeling = arguments['-vertfile']
-    else:
-        fname_vertebral_labeling = ""
-    if '-perslice' in arguments:
-        perslice = arguments['-perslice']
-    else:
-        perslice = param_default.perslice
-    if '-perlevel' in arguments:
-        perlevel = arguments['-perlevel']
-    else:
-        perlevel = 0
-    fname_normalizing_label = ''
-    if '-norm-file' in arguments:
-        fname_normalizing_label = arguments['-norm-file']
-    normalization_method = ''
-    if '-norm-method' in arguments:
-        normalization_method = arguments['-norm-method']
-    if '-fix-label' in arguments:
-        label_to_fix = arguments['-fix-label']
-    else:
-        label_to_fix = ''
-    if '-output-map' in arguments:
-        fname_output_metric_map = arguments['-output-map']
-    else:
-        fname_output_metric_map = ''
-    if '-mask-weighted' in arguments:
-        fname_mask_weight = arguments['-mask-weighted']
-    else:
-        fname_mask_weight = ''
-    # if 'discard_negative_values' in arguments:
-    discard_negative_values = int(arguments['-discard-neg-val'])
-    verbose = int(arguments.get('-v'))
+    arguments = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
+
+    overwrite = 0  # TODO: Not used. Why?
+    fname_data = sct.get_absolute_path(arguments.i)
+    path_label = arguments.f
+    method = arguments.method
+    fname_output = arguments.o
+    append_csv = arguments.append
+    combine_labels = arguments.combine
+    labels_user = arguments.l
+    adv_param_user = arguments.param  # TODO: Not used. Why?
+    slices_of_interest = arguments.z
+    vertebral_levels = arguments.vert
+    fname_vertebral_labeling = arguments.vertfile
+    perslice = arguments.perslice
+    perlevel = arguments.perlevel
+    fname_normalizing_label = arguments.norm_file  # TODO: Not used. Why?
+    normalization_method = arguments.norm_method  # TODO: Not used. Why?
+    label_to_fix = arguments.fix_label  # TODO: Not used. Why?
+    fname_output_metric_map = arguments.output_map  # TODO: Not used. Why?
+    fname_mask_weight = arguments.mask_weighted  # TODO: Not used. Why?
+    discard_negative_values = int(arguments.discard_neg_val)  # TODO: Not used. Why?
```

Comment:
Several of these arguments are just not used at all. Were they intended to be used? Abandoned? Should we keep them? 

I've marked them with comments for now.

Reply:
thanks for marking them. Yes, it's a long story. In short: some features were tried and then abandoned. But: some should be revived (perhaps)